File: new_mobile/lib/features/question/domain/entities/question.dart
import 'package:equatable/equatable.dart';

class Question extends Equatable {
  final String id;
  final String courseId;
  final String chapterId;
  final String subChapterId;
  final String description;
  final String choiceA;
  final String choiceB;
  final String choiceC;
  final String choiceD;
  final String answer;
  final String explanation;
  final bool isForQuiz;
  final bool isLiked;
  final String? relatedTopic;
  final String? userAnswer;

  const Question({
    required this.isLiked,
    required this.id,
    required this.courseId,
    required this.chapterId,
    required this.subChapterId,
    required this.description,
    required this.choiceA,
    required this.choiceB,
    required this.choiceC,
    required this.choiceD,
    required this.answer,
    required this.explanation,
    required this.isForQuiz,
    this.relatedTopic,
    this.userAnswer,
  });

  @override
  List<Object?> get props => [
        id,
        courseId,
        chapterId,
        subChapterId,
        description,
        choiceA,
        choiceB,
        choiceC,
        choiceD,
        answer,
        explanation,
        isForQuiz,
        relatedTopic,
        userAnswer,
      ];
}

File: new_backend/src/models/question.ts
import { Document, Schema, model } from "mongoose";

export interface IQuestion extends Document {
  courseId: Schema.Types.ObjectId;
  description: String;
  choiceA: String;
  choiceB: String;
  choiceC: String;
  choiceD: String;
  answer: String;
  relatedTopic: String;
  explanation: String;
  adminApproval: Boolean
  isForQuiz: Boolean;
  isForMock: Boolean;
  chapterId:  Schema.Types.ObjectId;
  subChapterId:  Schema.Types.ObjectId;
}

const QuestionSchema: Schema<IQuestion> = new Schema({
  chapterId:{
    type:Schema.Types.ObjectId,
    ref:'Chapter',
    required:[true,"chapter Id is required"]
  },
  courseId:{
    type:Schema.Types.ObjectId,
    ref:'Course',
    required:[true,"course Id is required"]
  },
  subChapterId:{
    type:Schema.Types.ObjectId,
    ref:'SubChapter',
    required:[true,"SubChapter Id is required"]
  },
  description: {
    type: String,
    required: [true, "question description is required"],
  },
  choiceA: {
    type: String,
    required: [true, "choice A is required"],
  },
  choiceB: {
    type: String,
    required: [true, "choice B is required"],
  },
  choiceC: {
    type: String,
    required: [true, "choice C is required"],
  },
  choiceD: {
    type: String,
    required: [true, "choice D is required"],
  },
  answer: {
    type: String,
    required: [true, "Answer is required"],
  },
  explanation: {
    type: String,
    required: [true, "explanation is required"],
  },
  relatedTopic: {
    type: String,
    default: "No related topic for now!"
  },
  isForQuiz: {
    type: Boolean,
    default: true
  },
  isForMock: {
    type: Boolean,
    default: false
  },
  adminApproval:{
    type: Boolean,
    default: true
  }
},
{
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt'
    }
  }
);

const Question = model<IQuestion>("Question", QuestionSchema);

export default Question;
